cpp/src/barretenberg/stdlib/recursion/verifier/verifier_turbo.test.cpp

```diff
@@ -1,516 +1,517 @@
+            inner_circuit_a, inner_circuit_b, outer_circuit, true, false, true);
         EXPECT_EQ(circuit_output.aggregation_state.public_inputs[0].get_value(), inner_inputs_a[0]);
         EXPECT_EQ(circuit_output.aggregation_state.public_inputs[1].get_value(), inner_inputs_a[1]);
         EXPECT_EQ(circuit_output.aggregation_state.public_inputs[2].get_value(), inner_inputs_a[2]);
 
-        EXPECT_EQ(inner_proof_result, barretenberg::fq12::one());
-
-        printf("composer gates = %zu\n", outer_composer.get_num_gates());
-
-        bool result = outer_composer.check_circuit();
-        EXPECT_EQ(result, true);
-        EXPECT_EQ(check_recursive_proof_public_inputs(
-                      outer_composer,
-                      outer_composer.composer_helper.crs_factory_->get_verifier_crs()->get_precomputed_g2_lines()),
-                  true);
-    }
-
-    static void test_inner_circuit()
-    {
-        if constexpr (!std::is_same<OuterComposer, plonk::StandardPlonkComposer>::value)
-            return; // We only want to run this test once (since it's not actually dependent on the typed test
-                    // parameter; which is the outer composer). We've only made it a typed test so that it can be
-                    // included in this test suite. So to avoid running this test identically 3 times, we escape all but
-                    // 1 permutation.
-
-        InnerComposer inner_composer = InnerComposer("../srs_db/ignition");
-        std::vector<barretenberg::fr> inner_inputs{ barretenberg::fr::random_element(),
-                                                    barretenberg::fr::random_element(),
-                                                    barretenberg::fr::random_element() };
-
-        create_inner_circuit(inner_composer, inner_inputs);
-
-        auto prover = inner_composer.create_prover();
-        auto verifier = inner_composer.create_verifier();
-        auto proof = prover.construct_proof();
-        auto verified = verifier.verify_proof(proof);
-        EXPECT_EQ(verified, true);
+        check_pairing(circuit_output);
+        check_recursive_verification_circuit(outer_circuit, true);
     }
 };
 
-typedef testing::Types<plonk::TurboPlonkComposer, plonk::StandardPlonkComposer> OuterComposerTypes;
+typedef testing::Types<plonk::StandardPlonkComposerHelper, plonk::TurboPlonkComposerHelper> OuterComposerTypes;
 
 TYPED_TEST_SUITE(stdlib_verifier_turbo, OuterComposerTypes);
 
+HEAVY_TYPED_TEST(stdlib_verifier_turbo, test_inner_circuit)
+{
+    TestFixture::test_inner_circuit();
+}
+
 HEAVY_TYPED_TEST(stdlib_verifier_turbo, recursive_proof_composition)
 {
     TestFixture::test_recursive_proof_composition();
 };
 
 HEAVY_TYPED_TEST(stdlib_verifier_turbo, double_verification)
 {
-    TestFixture::test_double_verification();
+    if constexpr (std::same_as<TypeParam, TurboPlonkComposerHelper>) {
+        TestFixture::test_double_verification();
+    } else {
+        // Test doesn't compile.
```

Comment:
Is this expected? Could be worth elaborating here

Reply:
Yeah, the old comment that the test was too heavy to run in general was misleading. It doesn't run because int he else branch there's a type clash due to how the test setup functions are written.